yarn/src/main/scala/org/apache/spark/deploy/yarn/security/CredentialUpdater.scala

```diff
@@ -107,8 +110,16 @@ private[spark] class ExecutorDelegationTokenUpdater(
     }
   }
 
+  private def getTimeOfNextUpdateFromFileName(credentialsPath: Path): Long = {
+    val name = credentialsPath.getName
```

Comment:
I wonder if using `name.split(SparkHadoopUtil.SPARK_YARN_CREDS_COUNTER_DELIM)` wouldn't make this method easier to read.